### PR TITLE
feat(cloud_firestore): Merging query snapshot with only changes

### PR DESCRIPTION
**Pull request by bswhite1**
Sorry, had to break this into a separate branch. Trying this again.

### CODE REVIEW

File: packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestoreMessageCodec.java [hunk 1]
```diff
@@ -301,6 +313,8 @@ private FirebaseFirestore readFirestoreInstance(ByteBuffer buffer) {
       FirebaseFirestore firestore = FirebaseFirestore.getInstance(app, databaseURL);
       firestore.setFirestoreSettings(settings);
 
+      firestore.setLoggingEnabled(false);
```
Comment (amagstadt) [inline packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestoreMessageCodec.java:316]: Why is this set to false?

Comment (bswhite1) [reply packages/cloud_firestore/cloud_firestore/android/src/main/java/io/flutter/plugins/firebase/firestore/FlutterFirebaseFirestoreMessageCodec.java:316]: It defaults to false. I just never remember where to turn it on, so thought. I would set it. This turns on firebase_android_sdk logging which is alot of logging.

File: packages/cloud_firestore/cloud_firestore_web/pubspec.yaml [hunk 2]
```diff
@@ -11,7 +11,12 @@ environment:
 
 dependencies:
   _flutterfire_internals: ^1.3.7
-  cloud_firestore_platform_interface: ^5.16.2
```
Comment (amagstadt) [inline packages/cloud_firestore/cloud_firestore_web/pubspec.yaml:14]: Are we truly using a platform that is a major version and **many** minor revisions old?

Comment (bswhite1) [reply packages/cloud_firestore/cloud_firestore_web/pubspec.yaml:14]: This is now a "tag" and it comes from cloud_firestore version here:
packages/cloud_firestore/cloud_firestore/pubspec.yaml

**Review by amagstadt — APPROVED** [review]

**Review by dparks23 — APPROVED** [review]

**Pull request merged**